sandbox/cgroup/monitor_test.go

```diff
@@ -0,0 +1,247 @@
+package cgroup_test
+
+import (
+	"os"
+	"path"
+	"time"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/sandbox/cgroup"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type monitorSuite struct {
+	testutil.BaseTest
+
+	tmp string
+	ch  chan string
+
+	inotifyWait time.Duration
+}
+
+var _ = Suite(&monitorSuite{})
+
+func (s *monitorSuite) SetUpTest(c *C) {
+	s.BaseTest.SetUpTest(c)
+
+	s.tmp = c.MkDir()
+	s.ch = make(chan string)
+	s.AddCleanup(func() { close(s.ch) })
+
+	s.calibrateInotifyDelay(c)
+}
+
+func (s *monitorSuite) calibrateInotifyDelay(c *C) {
+	folder1 := s.makeTestFolder(c, "folder1")
+	filelist := []string{folder1}
+	err := cgroup.MonitorDelete(filelist, "test1", s.ch)
+	c.Assert(err, IsNil)
+	var start time.Time
+	go func() {
+		start = time.Now()
+		err = os.Remove(folder1)
+		c.Assert(err, IsNil)
+	}()
+	<-s.ch
+	d := time.Now().Sub(start)
+	// On a modern machine the dureation "d" for inotify delivery is
+	// around 30-100µs so even the very conservative multiplication means
+	// the delay is typcially 3ms-10ms.
+	s.inotifyWait = 100 * d
+	switch {
+	case s.inotifyWait > 1*time.Second:
+		s.inotifyWait = 1 * time.Second
+	case s.inotifyWait < 10*time.Millisecond:
+		s.inotifyWait = 10 * time.Millisecond
+	}
+}
+
+func (s *monitorSuite) makeTestFolder(c *C, name string) (fullPath string) {
+	fullPath = path.Join(s.tmp, name)
+	err := os.Mkdir(fullPath, 0755)
+	c.Assert(err, IsNil)
+	return fullPath
+}
+
+func (s *monitorSuite) TestMonitorSnapBasicWork(c *C) {
+	folder1 := s.makeTestFolder(c, "folder1")
+	folder2 := s.makeTestFolder(c, "folder2")
+
+	filelist := []string{folder1}
+	err := cgroup.MonitorDelete(filelist, "test1", s.ch)
+	c.Assert(err, IsNil)
+
+	time.Sleep(s.inotifyWait)
+
+	err = os.Remove(folder2)
+	c.Assert(err, IsNil)
+
+	// Wait for two seconds to ensure that nothing spurious
+	// is received from the channel due to removing folder2
+	// Wait for two seconds to ensure that nothing spurious
+	// is received from the channel due to creating or
+	// removing folder3
+	select {
+	case event := <-s.ch:
+		c.Fatalf("unexpected channel read of event %q", event)
+	case <-time.After(2 * s.inotifyWait):
+	}
+
+	err = os.Remove(folder1)
+	c.Assert(err, IsNil)
+	event := <-s.ch
+	c.Assert(event, Equals, "test1")
+}
+
+func (s *monitorSuite) TestMonitorSnapTwoSnapsAtTheSameTime(c *C) {
+	folder1 := s.makeTestFolder(c, "folder1")
+	folder2 := s.makeTestFolder(c, "folder2")
+
+	filelist := []string{folder1, folder2}
+	err := cgroup.MonitorDelete(filelist, "test2", s.ch)
+	c.Assert(err, Equals, nil)
+
+	time.Sleep(s.inotifyWait)
+
+	folder3 := s.makeTestFolder(c, "folder3")
+
+	time.Sleep(s.inotifyWait)
+
+	err = os.Remove(folder3)
+	c.Assert(err, IsNil)
+
+	// Wait for two seconds to ensure that nothing spurious
+	// is received from the channel due to creating or
+	// removing folder3
+	select {
+	case event := <-s.ch:
+		c.Fatalf("unexpected channel read of event %q", event)
+	case <-time.After(2 * s.inotifyWait):
```

Comment:
Same as the previous select. I think this isn't necessary anymore
```suggestion
	case <-time.After(s.inotifyWait):
```